src/Queue/QueueServiceProvider.php

```diff
@@ -61,10 +62,16 @@ public function register()
         });
 
         $this->app->singleton(Worker::class, function ($app) {
+            /** @var Config $config */
+            $config = $app->make(Config::class);
+
             return new Worker(
-                new HackyManagerForWorker($app[Factory::class]),
```

Comment:
Is there a reason why we can use this now but couldn't before?

Reply:
Yes. We weren't able to override something. Not behind a computer but that's the bottom line.